apps/pipelines/models.py

```diff
@@ -9,3 +16,49 @@ class Pipeline(BaseTeamModel):
 
     class Meta:
         ordering = ["-created_at"]
+
+    def __str__(self):
+        return self.name
+
+    def invoke(self, input):
+        from apps.pipelines.graph import PipelineGraph
+
+        runnable = PipelineGraph.build_runnable_from_json(self.data)
+
+        pipeline_run = PipelineRun.objects.create(pipeline=self, status=PipelineRunStatus.RUNNING, log={"entries": []})
+        logging_callback = PipelineLoggingCallbackHandler(pipeline_run)
+        logging_callback.logger.info("Starting pipeline run")
+        try:
+            output = runnable.invoke(input, config=RunnableConfig(callbacks=[logging_callback]))
+        finally:
+            logging_callback.logger.info("Pipeline run finished")
+            logging_callback.pipeline_run.save()
+        return output
+
+
+class PipelineRunStatus(models.TextChoices):
+    RUNNING = "running", "Running"
+    SUCCESS = "success", "Success"
+    ERROR = "error", "Error"
+
+
+class PipelineRun(BaseModel):
+    pipeline = models.ForeignKey(Pipeline, on_delete=models.CASCADE, related_name="runs")
+    status = models.CharField()
```

Comment:
Is this `status` the same as the `status` on line 50?

Reply:
Good catch! I wanted the second one...